### ISSUE TITLE
Autocomplete for passwords is discouraged for security concerns (WCAG 2.1 Success Criterion 1.3.5)

### ISSUE THREAD

**Issue by RababGomaa** [w3c/wcag#2110]
**Issue**
Use of Autocompete `autocomplete="on"` attribute is described as a technique to meet 1.3.5 ref. [Using HTML 5.2 autocomplete attributes](https://www.w3.org/WAI/WCAG21/Techniques/html/H98)

 From security perspective, Autocompete is treated like browser-based passwords managers  which are considered a security concern. ref  [Password Managers-Security (ITSAP.30.025)](https://cyber.gc.ca/en/guidance/password-managers-security-itsap30025) .  The risk is higher for protected B material. Therefore, web teams are discouraged from using autocomplete and consequently the are failing on meeting [WCAG 2.1 1.3.5](https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose). 

 On the other hand, use of user password managers that require two‑factor authentication is a security recommendation.  However, two‑factor authentication has many ways of application and none is recommended for accessibility.

(Suggested labels : 1.3.5 Input purpose, WCAG 2.1)

**Recommendation**
 - Evaluation of  two factor authentication for user password fields for accessibility and recommend a method that works for people with disability and provide it as a WCAG technique. 
 - Consultation with ITSecurity teams would ensure that the technique in discussion is in alignment with security. 
 With this, development teams will have two options to choose from to satisfy WCAG 1.3.5 

fyi - Discussion happening within the Government of Canada, Digital Accessibility Toolkit [WCAG 2.1 Success Criterion 1.3.5: Identify Input Purpose: Passwords #54](https://github.com/gc-da11yn/gc-da11yn.github.io/discussions/54)

CC @shawnthompson

**Comment by wpcoc** [comment]
I have noticed this potential issue recently, although it hasn't directly affected my team yet. Working on WCAG support for one product, I can see autocomplete is recommended for passwords (`new-password` and `current-password`). But for another product that recently underwent a penetration test, the presence of an autocomplete attribute caused a low-severity flag to be raised on the pentest report ([example detail](https://portswigger.net/kb/issues/00500800_password-field-with-autocomplete-enabled)). Sadly, we **have to** favour the security report over the accessibility guidelines, especially due to the mentioned possible knock-on impact on PCI compliance, so there is no opportunity for discussion on this for us. I'm hopeful that an agreement can be reached with regards to this issue. In the meantime, I can only hope that the relevant accessibility tools will still operate as expected when faced with password fields with `autocomplete="off"`

**Comment by patrickhlauke** [comment]
Coming back to this, I would respectfully argue that the pentest flag is misguided. Trying to force users (by trying to prevent their browsers from not offering to store login credentials) not to use password managers (be they built-in, or integrated into their browser through extensions) does nothing other than forcing users to either choose weaker passwords they can remember, or at the very least cause inconvenience and friction.

The throwaway claim at the end that attackers may be able to then somehow compromise the browser's password storage is FUDdy as well. If an attacker manages to inject a cross-site script into a login page, they'll get the password either way ... either when it's autocompleted, or when the user actually enters it manually to log in.

**Comment by patrickhlauke** [comment]
Closing this, as the original question (or rather, the source of it ... that "pentest documentation") appears to confuse the underlying aspects related to password managers and autocomplete attributes.
